modules/budgets/app/controllers/budgets_controller.rb

```diff
@@ -136,11 +137,21 @@ def update
   end
 
   def destroy
+    budget = Budget.find(params[:id])
+    if budget.work_packages.any?
+      redirect_to destroy_info_budget_path(budget)
+      return
+    end
+
     @budgets.each(&:destroy)
     flash[:notice] = t(:notice_successful_delete)
     redirect_to action: 'index', project_id: @project
   end
 
+  def destroy_info
+   @possible_other_budgets = get_all_project_budgets - [@budget]
```

Comment:
Here I'd prefer you use a query to exclude the current one: `@project.budgets.where.not(id: @budget.id)`